app/resources/v1/users.js

```diff
@@ -381,16 +386,40 @@ exports.put = async function (req, res) {
     return
   }
 
+  // Is requesting user logged in?
   if (user.login_tokens.indexOf(req.loginToken) === -1) {
     res.status(401).end()
     return
   }
 
-  user.data = body.data || user.data
-  user.save().then(user => {
-    res.status(204).end()
-  }).catch(err => {
-    logger.error(err)
-    res.status(500).send('Could not update user information.')
-  })
+  // Find target user
+  const targetUserId = req.params.user_id
+  let targetUser
+  if (userId !== targetUserId) {
+    try {
+      targetUser = await User.findOne({ id: targetUserId })
+    } catch (err) {
+      logger.error(err)
+      res.status(500).send('Error finding user.')
+    }
+
+    if (!targetUser) {
+      res.status(404).send('User not found.')
+      return
+    }
+  } else {
+    targetUser = user
+  }
+
+  if (userId === targetUserId || user.role.includes('ADMIN')) {
```

Comment:
Looks like travisci is logging an error if `user.role` does not exist? Should we make the default value for `user.role = ["user"]` or an empty array?

Reply:
Hmm! We should probably make `user.role` an empty array so that it's always there, even if it has no information.

Should it be `user.roles` (plural) because an array implies any number of roles?

Reply:
Oh true. Yes, it would make more sense as `user.roles`

Reply:
OK! Pushed that change, I'll update staging database as well.